libdnf/module/module_sack.cpp

```diff
@@ -478,6 +482,149 @@ std::pair<std::vector<std::vector<std::string>>, ModuleSack::ModuleErrorType> Mo
+
+    libdnf::rpm::PackageQuery base_query(base);
+    base_query.filter_provides({"system-release"});
+    base_query.filter_latest_evr();
+
+    // try to detect platform id from available packages
+    libdnf::rpm::PackageQuery available_query(base_query);
+    available_query.filter_available();
+    auto available_provides = get_strings_from_provide(available_query, "base-module");
+    if (available_provides.size() == 1) {
+        auto & query_platform_id = *available_provides.begin();
+        auto parsed_platform_id = parse_platform_id_from_string(query_platform_id);
+        if (!parsed_platform_id.first.empty() && !parsed_platform_id.second.empty()) {
+            return parsed_platform_id;
+        } else {
+            base->get_logger()->warning(
+                "Invalid format of module platform queried from available packages: {}", query_platform_id);
+        }
+    } else {
+        base->get_logger()->warning("Multiple module platforms provided by available packages");
+    }
+
+    // try to detect platform id from installed packages
+    libdnf::rpm::PackageQuery installed_query(base_query);
+    installed_query.filter_installed();
+    auto installed_provides = get_strings_from_provide(installed_query, "base-module");
+    if (installed_provides.size() == 1) {
+        auto & query_platform_id = *installed_provides.begin();
+        auto parsed_platform_id = parse_platform_id_from_string(query_platform_id);
+        if (!parsed_platform_id.first.empty() && !parsed_platform_id.second.empty()) {
+            return parsed_platform_id;
+        } else {
+            base->get_logger()->warning(
+                "Invalid format of module platform queried from installed packages: {}", query_platform_id);
+        }
+    } else {
+        base->get_logger()->warning("Multiple module platforms provided by installed packages");
+    }
+
+    // try to detect platform id from release files
+    // TODO(jkolarik): Create constants for paths
+    std::vector<std::string> os_release_paths{"etc/os-release", "usr/lib/os-release"};
+    auto & installroot = base->get_config().installroot().get_value();
+    for (auto & os_release_path : os_release_paths) {
+        std::string full_path = std::filesystem::canonical(installroot) / os_release_path;
+        std::pair<std::string, std::string> parsed_platform_id;
+
+        try {
+            parsed_platform_id = parse_platform_id_from_file(full_path);
+        } catch (const std::exception & except) {
+            base->get_logger()->debug(
+                "Detection of module platform in {} failed: {}", full_path, std::string(except.what()));
+        }
+
+        if (!parsed_platform_id.first.empty() && !parsed_platform_id.second.empty()) {
+            return parsed_platform_id;
+        } else {
+            base->get_logger()->warning("Error when parsing platform ID from release path: {}", full_path);
+        }
+    }
+
+    throw std::runtime_error("No valid Platform ID detected");
```

Comment:
I am sorry but I have a feeling that terminating DNF here is not a proper action. Platform ID is not an essential part of modularity design. It is a hack from Modular Build System, that we have to handle somehow. Do you have some reason to terminate DNF when platform is not around?

Reply:
No, I don't have any valid reason, it's just my lack of knowledge about how the platform ID is internally handled by modularity and what are all the consequences of not having the virtual platform solvable present.

Reply:
I would suggest to replace it by a warning, info, or debug message. May be debug message would be a better option.